components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/utils/AnalyticsPublishServiceUtils.java

```diff
@@ -0,0 +1,301 @@
+	 */
+	private void writePublishTimeToRegistry(
+			List<HistoricProcessInstance> historicProcessInstanceList) {
+		log.debug("Start writing last completed process instance publish time...");
+		Date lastProcessInstancePublishTime =
+				historicProcessInstanceList.get(historicProcessInstanceList.size() - 1)
+				                           .getStartTime();
+		try {
+			PrivilegedCarbonContext context = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+			Registry registry = context.getRegistry(RegistryType.SYSTEM_GOVERNANCE);
+			Resource resource;
+			if (!registry.resourceExists(AnalyticsPublisherConstants.PROCESS_RESOURCE_PATH)) {
+				resource = registry.newResource();
+				resource.addProperty(AnalyticsPublisherConstants.LAST_PROCESS_INSTANCE_PUBLISH_TIME,
+				                     String.valueOf(lastProcessInstancePublishTime));
+				registry.put(AnalyticsPublisherConstants.PROCESS_RESOURCE_PATH, resource);
+			} else {
+				resource = registry.get(AnalyticsPublisherConstants.PROCESS_RESOURCE_PATH);
+				resource.setProperty(AnalyticsPublisherConstants.LAST_PROCESS_INSTANCE_PUBLISH_TIME,
+				                     String.valueOf(lastProcessInstancePublishTime));
+				registry.put(AnalyticsPublisherConstants.PROCESS_RESOURCE_PATH, resource);
+			}
+			log.debug("End of writing last completed process instance publish time...");
+		} catch (RegistryException e) {
+			String errMsg = "Registry error while writing the process instance publish time.";
+			log.error(errMsg, e);
+		}
+	}
+
+	/**
+	 * Write last completed task instance end time to carbon registry
+	 *
+	 * @param historicTaskInstanceList List of historic task instances
+	 */
+	private void writeTaskEndTimeToRegistry(List<HistoricTaskInstance> historicTaskInstanceList) {
+		log.debug("Start writing last completed task instance end time...");
+		Date lastTaskInstanceDate =
+				historicTaskInstanceList.get(historicTaskInstanceList.size() - 1).getEndTime();
+		try {
+			PrivilegedCarbonContext privilegedContext =
+					PrivilegedCarbonContext.getThreadLocalCarbonContext();
+			Registry registry = privilegedContext.getRegistry(RegistryType.SYSTEM_GOVERNANCE);
+			Resource resource;
+			if (!registry.resourceExists(AnalyticsPublisherConstants.TASK_RESOURCE_PATH)) {
+				resource = registry.newResource();
+				resource.addProperty(AnalyticsPublisherConstants.LAST_TASK_INSTANCE_END_TIME,
+				                     String.valueOf(lastTaskInstanceDate));
+				registry.put(AnalyticsPublisherConstants.TASK_RESOURCE_PATH, resource);
+			} else {
+				resource = registry.get(AnalyticsPublisherConstants.TASK_RESOURCE_PATH);
+				resource.setProperty(AnalyticsPublisherConstants.LAST_TASK_INSTANCE_END_TIME,
+				                     String.valueOf(lastTaskInstanceDate));
+				registry.put(AnalyticsPublisherConstants.TASK_RESOURCE_PATH, resource);
+			}
+			log.debug("End of writing last completed task instance end time...");
```

Comment:
this seems to be a generic debug log ... does it make sense to have more information?